src/components/profile.ts

```diff
@@ -0,0 +1,153 @@
+): userCustomization => {
+  if (!(Object.keys(configMaps) as Array<keyof typeof configMaps>).includes(customization)) {
+    return {
+      reason: `Invalid customization selection. Must be one of **${(Object.keys(configMaps) as Array<keyof typeof configMaps>).map((key) => ' ' + key)}**`};
+  }
+  let parsedDescription = description;
+  switch (customization) {
+    case 'birthdate':
+      // convert first letter to capital in case the user doesnt use proper capitalization
+      parsedDescription = description.toLowerCase();
+      parsedDescription = parsedDescription[0].toUpperCase() + parsedDescription.slice(1);
+      if (!validBirthdates.includes(parsedDescription)) {
+        return { reason: 'Invalid birthdate. Must be one of : ' + validBirthdates.join(', ') + '.' };
+      }
+      break;
+    case 'term':
+      parsedDescription = description.length == 2 ? description.toUpperCase() : description;
+      if (!validTerms.includes(parsedDescription)) {
+        return { reason: 'Invalid term. Must be one of : ' + validTerms.join(', ') + '.' };
+      }
+      break;
+    case 'faculty':
+      // convert first letter to capital in case the user doesnt use proper capitalization
+      parsedDescription = description.toLowerCase();
+      parsedDescription = parsedDescription[0].toUpperCase() + parsedDescription.slice(1);
+      if (!validFaculties.includes(parsedDescription)) {
+        return { reason: 'Invalid faculty. Must be one of : ' + validFaculties.join(', ') + '.' };
+      }
+      break;
+    case 'year':
+      if (!validYears.includes(parseInt(description))) {
+        return { reason: 'Invalid year. Must be between of : ' + validYears.join(', ') + '.' };
+      }
+      break;
+  }
+  return { reason: 'valid', parsedDescription };
+};
+
+export const getUserProfileById = async (userId: string): Promise<UserProfile | undefined> => {
+  const db = await openDB();
+  return await db.get('SELECT * FROM user_profile_table WHERE user_id = ?', userId);
+};
+
+export const editUserProfileById = async (userId: string, data: UserProfile): Promise<void> => {
+  const db = await openDB();
+  const res = await db.get(`SELECT COUNT(*) as found FROM user_profile_table where user_id = ?`, userId);
+  const user = res.found == 1;
+  let query;
+  if (user) {
+    query = `UPDATE user_profile_table SET last_updated=CURRENT_TIMESTAMP, `;
+    for (const [customization, description] of Object.entries(data)) {
+      query = query.concat(`${customization}=IFNULL('${description}', ${customization}) `);
+    }
+    query = query.concat('WHERE user_id = ?');
+    await db.run(query, userId);
+  } else {
+    query = `
+            INSERT INTO user_profile_table 
+            (user_id, last_updated,
+            `;
+    for (const customization of Object.keys(data)) {
+      query = query.concat(`${customization})`);
+    }
+    query = query.concat(`VALUES (?, CURRENT_TIMESTAMP, ?)`);
+    await db.run(query, userId, ...Object.values(data));
```

Comment:
and this, I guess.
or frankly, at the start of the file, note somewhere that data only has one key-val pair